llarp/service/endpoint.hpp

```diff
@@ -59,16 +59,16 @@ namespace llarp
     };
     using ConvoEventListener_ptr = std::shared_ptr< IConvoEventListener >;
 
+    /// minimum interval for publishing introsets
+    static constexpr auto INTROSET_PUBLISH_INTERVAL =
+        path::default_lifetime / 4;
```

Comment:
Is integer truncation going to behave differently / less predictably with `std::chrono`?

```cpp
int seconds = 15 / 4
```
is very different from
```cpp
int ms = 1500 / 4
```

`/ 4` just may not be a very safe way to do this in general...

Reply:
it shouldn't be that bad right?

Reply:
as long as it's a multiple of 4 :laughing:

Reply:
`default_lifetime` is 600s, isn't it?

Reply:
No, it's 10min.  so we probably want something like `std::chrono::seconds{path::default_lifetime} / 4`

Reply:
```c++
#include <chrono>
#include <iostream>

using namespace std::chrono_literals;

int main(int argc, char** argv) {
    auto d1 = 10min / 4;
    std::cout << "d1 = " << d1.count() << std::endl;

    auto d2 = std::chrono::duration_cast<std::chrono::milliseconds>(d1);
    std::cout << "d2 = " << d2.count() << std::endl;

    return 0;
}
```
output:
```
d1 = 2
d2 = 120000
```

So there's definitely a caveat here with the time unit used for `time_point` or `duration`. Probably to my earlier point: division may best be avoided, and when it's not, consider carefully how truncation will play out.

Reply:
It might be a good idea to standardize on a time unit (milliseconds or microseconds) and use those when defining types. Obviously this would still allow for passing other types, but would cut down on the chances of division and/or `count()` giving unintended results.

Ex:

```c++
struct Foo {
  std::chrono::milliseconds m_timeout;
  void setTimeout(std::chrono::milliseconds timeout);
};

Foo foo;
foo.setTimeout(5s); // still able to use literals or durations with arbitrary time units
```

Reply:
as i understand it we chose milliseconds long ago as the time unit. has that changed or no?